### ISSUE TITLE
nomodeset required on Linux v5.15.99 with i915 graphics

### ISSUE THREAD

**Issue by paulwalker-dev** [NixOS/nixpkgs#220909]
### Describe the bug
Systemd seems to hang during boot without "nomodeset" being set as a kernel parameter under intel graphics

### Steps To Reproduce
1. Use intel integrated graphics.
2. Boot from Linux v5.15.99 with a graphical target set.

### Expected behavior
Modeset should work and allow Display Managers like GDM to appear.
This also applies to X11 and Wayland window/desktop managers.

### Notify maintainers
@Ma27

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.97, NixOS, 22.11 (Raccoon), 22.11.20230310.824f886`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by epixtm** [comment]
Can confirm, updated a T420 today and was unable to boot the 5.15.99 kernel. Other people are reporting [failing to boot on this version aswell w/ i915 graphics](https://gitlab.alpinelinux.org/alpine/aports/-/issues/14704). Looks like `5.15.101` has already been pushed to undo the regression.

**Comment by sigprof** [comment]
Can reproduce the problem on a desktop i3-10105 too.  Grabbing the 5.15.101 kernel from the `nixos-22.11-small` channel fixes things while waiting for the full `nixos-21.11` to update:
```
    nixpkgs-small.url = "github:NixOS/nixpkgs/nixos-22.11-small";
```
```
    boot.kernelPackages = inputs.nixpkgs-small.legacyPackages.${pkgs.system}.linuxPackages;
```

**Comment by Ma27** [comment]
Fixed in 5.15.101 which is already backported (though not in the channel yet), so we can close this.
